config/crd/bases/approuting.kubernetes.azure.com_nginxingresscontrollers.yaml

```diff
@@ -104,6 +104,26 @@ spec:
                   to the NGINX Ingress Controller's Service. Common annotations will
                   be from the Azure LoadBalancer annotations here https://cloud-provider-azure.sigs.k8s.io/topics/loadbalancer/#loadbalancer-annotations
                 type: object
+              scaling:
+                description: Scaling defines configuration options for how the Ingress
+                  Controller scales
+                properties:
+                  maxReplicas:
+                    description: MaxReplicas is the upper limit for the number of
+                      Ingress Controller replicas. It defaults to 100 pods.
+                    format: int32
+                    minimum: 1
+                    type: integer
+                  minReplicas:
+                    description: MinReplicas is the lower limit for the number of
+                      Ingress Controller replicas. It defaults to 2 pods.
+                    format: int32
+                    minimum: 1
+                    type: integer
+                type: object
+                x-kubernetes-validations:
+                - rule: (!has(self.minReplicas)) || (!has(self.maxReplicas)) || (self.minReplicas
```

Comment:
Just a question since I'm not familiar with this: why does the rule check if minReplicas or maxReplicas are not set? Also, is there any limit for maxReplica count?

Reply:
they are pointer values so it's possible for a user to not have them set or only one to be set. in CEL (the validation language) there's no default values so if we were to compare something null vs a defined thing it would panic.

there is no limit for max replica count. this matches the form of the hpa which this field eventually flows into

Reply:
if customers want scale... we'll give em scale! 😆

Reply:
also note that this flows down to an HPA which will only scale to the max if that number is actually needed

Reply:
thanks, that makes sense!